ragna/core/_rag.py

```diff
@@ -188,7 +202,14 @@ async def prepare(self) -> Message:
                 detail=RagnaException.EVENT,
             )
 
-        await self._run(self.source_storage.store, self.documents)
+        if list[Document] in inspect.signature(self.source_storage.store).parameters.values():
+            await self._run(self.source_storage.store, self.documents)
+        else:
+            # Here we need to generate the list of embeddings
+            chunks = self.chunking_model.chunk_documents(self.documents)
+            embeddings = self.embedding_model.embed_chunks(chunks)
```

Comment:
The source storage should also be able to just request chunks. Meaning, we have three distinct cases and cannot group these two. However, if we split this PR as suggested above, this distinction will only come in the follow-up PR.

Reply:
So something like?
``` python
        if type(self.source_storage).__ragna_input_type__ == Document:
            await self._run(self.source_storage.store, self.documents)
        else:
            chunks = self.chunking_model.chunk_documents(documents=self.documents)
            if type(self.source_storage).__ragna_input_type__ == Chunk:
                await self._run(self.source_storage.store, chunks)
            else:
                embeddings = self.embedding_model.embed_chunks(chunks)
                await self._run(self.source_storage.store, embeddings)
```